tests/test_pagination.py

```diff
@@ -475,52 +477,75 @@ class TestCursorPagination:
     """
+    class MockObject(object):
+        def __init__(self, idx):
+            self.created = idx
 
-    def setup(self):
-        class MockObject(object):
-            def __init__(self, idx):
-                self.created = idx
-
-        class MockQuerySet(object):
-            def __init__(self, items):
-                self.items = items
-
-            def filter(self, created__gt=None, created__lt=None):
-                if created__gt is not None:
-                    return MockQuerySet([
-                        item for item in self.items
-                        if item.created > int(created__gt)
-                    ])
-
-                assert created__lt is not None
-                return MockQuerySet([
+    class MockQuerySet(object):
+        def __init__(self, items):
+            self.items = list(items)
+
+        def filter(self, created__gt=None, created__lt=None):
+            if created__gt is not None:
+                return type(self)([
                     item for item in self.items
-                    if item.created < int(created__lt)
+                    if item.created > int(created__gt)
                 ])
 
-            def order_by(self, *ordering):
-                if ordering[0].startswith('-'):
-                    return MockQuerySet(list(reversed(self.items)))
-                return self
+            assert created__lt is not None
+            return type(self)([
+                item for item in self.items
+                if item.created < int(created__lt)
+            ])
+
+        def order_by(self, *ordering):
+            if ordering[0].startswith('-'):
+                return type(self)(list(reversed(self.items)))
+            return self
+
+        def __getitem__(self, sliced):
+            return self.items[sliced]
+
+    class ExamplePagination(pagination.CursorPagination):
+        page_size = 5
+        ordering = 'created'
+
+    class CustomCursorPagination(ExamplePagination):
+        cursor_query_param = 'since'
+        reverse_query_param = 'before'
+
+        def decode_cursor(self, request):
+            rev = False
+            if self.reverse_query_param in request.query_params:
+                rev = True
+                query_param = self.reverse_query_param
+            elif self.cursor_query_param in request.query_params:
+                query_param = self.cursor_query_param
+            else:
+                return
+            return Cursor(0, rev, request.query_params[query_param])
+
+        def encode_cursor(self, cursor):
+            if cursor.reverse:
+                query_param = self.reverse_query_param
+                base_url = remove_query_param(self.base_url, self.cursor_query_param)
+            else:
+                query_param = self.cursor_query_param
+                base_url = remove_query_param(self.base_url, self.reverse_query_param)
+            return replace_query_param(base_url, query_param, cursor.position)
```

Comment:
I don't think we really need to test the hooks themselves. They'll be sufficiently covered by the existing test suite, so I think this PR would be better off without any test changes, purely for simplicities sake.